sfa_dash/templates/forms/cdf_forecast_group_form.html

```diff
@@ -6,17 +6,22 @@
 {% set form_data = {} %}
 {% endif %}
 {% block fields %}
-<input type="hidden" name="site_id" value="{{ site_metadata['site_id'] }}">
 <div class="form-element">
   {{ form.input('Name', 'text', 'name', 'Name for the Forecast',
                 required=True, form_data=form_data) }}
 </div>
+{# if a site is defined, allow users to select a variable, else fetch the
+   variable from aggregate metadata #}
+{% if site_metadata is defined %}
```

Comment:
is there a way site_metadata won't be defined. For example, if the user does not have access to the site metadata but tried to clone a forecast? And then a variable couldn't be set

Reply:
Users should be directed away from the form if the associated site/aggregate is inaccessible before the template is rendered. This is to enforce the rule that a user must have read permissions on a site/aggregate to create observations or forecasts on it. The situation where neither site_metadata or aggregate_metadata will not be set should only be when creating or cloning a site or aggregate.
This does make me realize I need to think about controlling the display of a clone button, which I haven't added yet, because simple read permissions will not suffice.